### PR TITLE
Bump googletest version to v1.11.0

### LINKED ISSUE #60865: `clang-tidy` CI: `GTEST_SKIP()` identifier cannot be found
When running `clang-tidy` on `torch/csrc/deploy/test_deploy_gpu.cpp`, CI reports an error `GTEST_SKIP()` not found

## To Reproduce

1. Run `tools/linter/clang-tidy.py` on `torch/csrc/deploy/test_deploy_gpu.cpp`

## Additional Context

Grepping for `GTEST_SKIP()` inside our checkout of `googletest` does not return any results. Further, `torch/csrc/deploy/test_deploy_gpu.cpp` is not being used during testing ([see here](https://github.com/pytorch/pytorch/blob/9c4eec2a2db26ab96872edd9c08980d98016d5cc/torch/csrc/deploy/CMakeLists.txt#L18-L20)).

### PR DESCRIPTION
**Pull request by 1ntEgr8**
This PR bumps the `googletest` version to v1.11.0. 

To facilitate this change, `CAFFE2_ASAN_FLAG` and `CAFFE2_TSAN_FLAG` are divided into corresponding compiler and linker variants. This is required because `googletest v1.11.0` sets the `-Werror` flag. The `-pie` flag is a linker flag, and passing it to a compiler invocation results in a `-Wunused-command-line-argument` warning, which in turn will cause `googletest` to fail to build with ASAN.

Fixes #60865

### CODE REVIEW

File: cmake/MiscCheck.cmake [hunk 1]
```diff
@@ -319,21 +319,23 @@ endif()
...
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${CAFFE2_ASAN_FLAG}")
-  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${CAFFE2_ASAN_FLAG}")
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${CAFFE2_ASAN_FLAG}")
+  set(CAFFE2_ASAN_COMPILER_FLAGS "-fsanitize=address -fPIE")
+  set(CAFFE2_ASAN_LINKER_FLAGS "-pie")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CAFFE2_ASAN_COMPILER_FLAGS}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CAFFE2_ASAN_COMPILER_FLAGS}")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${CAFFE2_ASAN_LINKER_FLAGS}")
+  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${CAFFE2_ASAN_LINKER_FLAGS}")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${CAFFE2_ASAN_LINKER_FLAGS}")
```
Comment (malfet) [inline cmake/MiscCheck.cmake:328]: Are those relevant to this PR? If not, please move them to a separate PR

**Review by malfet — CHANGES_REQUESTED** [review]
Looks like you have some unrelated changes in this PR. Please remove them
If they are related, please add an explanation to PR description, why this is a prerequisite for version update

File: cmake/MiscCheck.cmake (hunk 1 above)
Comment (1ntEgr8) [reply cmake/MiscCheck.cmake:328]: Yes, they are relevant. The ASAN tests were failing without this change

**Review by malfet — APPROVED** [review]
